### PR TITLE
native: sync fixes

### PR DESCRIPTION
**Pull request by dnbrwstr**
- Fixes TLON-1900 (optimistic updates)
- Fixes TLON-1972 ("last post missing" weirdness)
- Fixes TLON-1985 (Chat list's initial flickering)

Might fix TLON-1939

### CODE REVIEW

File: packages/shared/src/db/queries.ts [hunk 1]
```diff
@@ -1013,19 +1013,13 @@ export const insertChannels = createWriteQuery(
       return;
     }
 
-    logger.log(
```
Comment (dnbrwstr) [inline packages/shared/src/db/queries.ts:1016]: Why

File: packages/shared/src/db/queries.ts [hunk 2]
```diff
@@ -615,8 +615,8 @@ export const addChatMembers = createWriteQuery(
     },
     ctx: QueryCtx
   ) => {
-    return ctx.db.transaction(async (tx) => {
-      await tx
+    return withTransactionCtx(ctx, () => {
+      return ctx.db
```
Comment (dnbrwstr) [inline packages/shared/src/db/queries.ts:619]: txCtx

File: packages/shared/src/db/queries.ts [hunk 3]
```diff
@@ -1476,15 +1470,6 @@ export const insertStandalonePosts = createWriteQuery(
   async (posts: Post[], ctx: QueryCtx) => {
     withTransactionCtx(ctx, async (txCtx) => {
       await insertPosts(posts, txCtx);
-      for (const post of posts) {
```
Comment (dnbrwstr) [inline packages/shared/src/db/queries.ts:1479]: Also rename insertStandalone to insertLatest

File: packages/shared/src/store/sync.ts [hunk 4]
```diff
@@ -518,36 +522,34 @@ export async function syncChannelWithBackoff({
 }: {
   channelId: string;
 }): Promise<boolean> {
+  async function isStillPending() {
+    return (
+      currentPendingMessageSyncs.has(channelId) &&
```
Comment (dnbrwstr) [inline packages/shared/src/store/sync.ts:527]: Nop

**Review by latter-bolden — APPROVED** [review]
We walked through this together, feedback captured in Dan's comments. Good to merge once those are resolved

**Pull request merged**